blackbird_python/blackbird/listener.py

```diff
@@ -308,7 +308,19 @@ def exitStatement(self, ctx: blackbirdParser.StatementContext):
             op_args, op_kwargs = _get_arguments(ctx.arguments())
 
             # convert any sympy expressions into regref transforms
-            op_args = [RegRefTransform(i) if isinstance(i, sym.Expr) else i for i in op_args]
+            for idx, a in enumerate(op_args):
+                if isinstance(a, sym.Expr):
+                    p = [str(i)[0] == 'q' for i in a.free_symbols]
+
+                    if any(p):
+                        op_args[idx] = RegRefTransform(a)
+
+            for k, v in op_kwargs.items():
```

Comment:
This is because I originally attempted to match what happens in Strawberry Fields; in Strawberry Fields, both of the following are valid:
```python
MeasureX | q[0]
Sgate(q[0]) | q[1]
Sgate(r=q[0]) | q[1]
```
It does make it more complicated though.